app/com/lynxanalytics/biggraph/frontend_operations/GraphComputationOperations.scala

```diff
@@ -85,6 +85,19 @@ class GraphComputationOperations(env: SparkFreeEnvironment) extends ProjectOpera
     }
   })
 
+  register("Find k-core decomposition")(new ProjectTransformation(_) {
+    params ++= List(Param("name", "Attribute name", defaultValue = "core"))
+    def enabled = project.hasEdgeBundle
+    def apply() = {
+      val name = params("name")
+      // A directed graph just gets turned into an undirected graph.
+      // We can skip that and just build an undirected graph.
+      val core = graph_operations.NetworKitComputeAttribute.run(
+        "CoreDecomposition", es, Map("directed" -> false))
+      project.newVertexAttribute(name, core, help)
+    }
+  })
```

Comment:
I'm just wondering, shouldn't this create a segmentation instead of an attribute?

Reply:
> I'm just wondering, shouldn't this create a segmentation instead of an attribute?

I don't think it's clear. With an attribute you can immediately put in a filter and get a core, or color a visualization by the core number. What would be easier with a segmentation? Getting the average of some attribute over a core? But that's a bit weird. The vertices in a core are not really related to each other. I don't think there are many cases where you would benefit from that.

But of course there's nothing stopping us from outputting a segmentation in addition to the attribute! Or make it a checkbox. NetworKit outputs the results as a segmentation (Partition) too. I just haven't written the code to turn a NetworKit Partition into a LynxKite segmentation yet. But that's certainly coming.